libopenage/renderer/renderer.cpp

```diff
@@ -0,0 +1,35 @@
+// Copyright 2015-2019 the openage authors. See copying.md for legal info.
+
+#include "renderer.h"
+
+namespace openage {
+namespace renderer {
+
+RenderPass::RenderPass(std::vector<Renderable> renderables, RenderTarget const* target)
+:renderables(renderables), target(target), is_optimised(false) {}
+
+std::vector<Renderable> RenderPass::get_renderables() const {
+  return this->renderables;
+}
+
+void RenderPass::set_renderables(std::vector<Renderable> renderables) {
+  this->renderables = renderables;
```

Comment:
Should this set `is_optimised = false`?

Reply:
I was also thinking about that. It should definitely not set `is_optimised = false` because it's only used to replace the renderables with the ones that are optimized. For some reason `std::sort` refuses to directly modify the member value. I was thinking that maybe we should name it `set_optimized_renderables` to avoid undesired use.

Reply:
But you set `is_optimised` after setting the renderables, so even if it is initially reset to `false`, it should be fine, no?

If you'd like to make some functionality only available to the renderer but not to users, you could follow the design of other components (e.g. `GlShaderProgram`) in which there is a base abstract class `Renderable` and a derived concrete `GlRenderable` which adds on the renderer-only functionality.

Reply:
>But you set is_optimised after setting the renderables

Yes, you're right

>base abstract class RenderPass and a derived concrete GlRenderPass

Sounds like the right thing to do!
So, I did it. Take a look

Reply:
Hey! Sorry about the late response, this looks good!